### PR TITLE
Add 'make package' target for building source and binary distributions

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Add the `make package` target for building packages.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: .github/workflows/publish-to-pypi.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
 
     - name: Build source and wheel distributions
       run: |
-        python setup.py sdist bdist_wheel
+        make package
```
Comment (weiji14) [inline .github/workflows/publish-to-pypi.yml:50]: I almost want to keep this command as a pure python command, that way people without knowledge of `make` (e.g. Windows users) can understand what is happening as well.

File: Makefile [hunk 2]
```diff
@@ -26,6 +27,9 @@ help:
 install:
 	pip install --no-deps -e .
 
+package:
+	python setup.py sdist bdist_wheel
```
Comment (weiji14) [inline Makefile:31]: Just wondering, when do we want to use this `make package` command? And what is the different between `python setup.py sdist bdist_wheel` and `python setup.py sdist --formats=zip` used in 'ci_tests.yaml'?

File: .github/workflows/publish-to-pypi.yml (hunk 1 above)
Comment (seisman) [reply .github/workflows/publish-to-pypi.yml:50]: We're heavily using `make` commands in our workflows. The reason for replacing `python setup.py` with `make` is to keep it simple when we switch to other tools for building packages.

File: Makefile (hunk 2 above)
Comment (seisman) [reply Makefile:31]: > Just wondering, when do we want to use this `make package` command? 

I was trying to address issue #904, so had to run the commands many times. `make pacakge` is easier to type.

> And what is the different between `python setup.py sdist bdist_wheel` and `python setup.py sdist --formats=zip` used in 'ci_tests.yaml'?

`sdist` is the source distribution, and `bdist_wheel` is the binary distribution (in wheel format). They are uploaded to PyPI (https://pypi.org/project/pygmt/#files). 

In `ci_tests.yaml`, we need to install PyGMT, so we only need the source distribution (`sdist`).

Comment (weiji14) [reply Makefile:31]: > > Just wondering, when do we want to use this `make package` command?
> 
> I was trying to address issue #904, so had to run the commands many times. `make pacakge` is easier to type.

Ok, fair enough. I would just use the [CTRL+R reverse-i-search](https://www.chriswrites.com/search-terminal-history/) command to save on typing an old command but `make package` works too.

> > And what is the different between `python setup.py sdist bdist_wheel` and `python setup.py sdist --formats=zip` used in 'ci_tests.yaml'?
> 
> `sdist` is the source distribution, and `bdist_wheel` is the binary distribution (in wheel format). They are uploaded to PyPI (https://pypi.org/project/pygmt/#files).
> 
> In `ci_tests.yaml`, we need to install PyGMT, so we only need the source distribution (`sdist`).

Got it, thanks for the explanation.

File: .github/workflows/publish-to-pypi.yml (hunk 1 above)
Comment (weiji14) [reply .github/workflows/publish-to-pypi.yml:50]: Let's not jinx things, I'm hoping Github Actions will last for a couple more years at least! Or maybe you mean switching to use `poetry` or something? EIther way, ok with using `make` here even though it's biased towards Linux/macOS.

**Review by weiji14 — APPROVED** [review]

File: .github/workflows/publish-to-pypi.yml (hunk 1 above)
Comment (seisman) [reply .github/workflows/publish-to-pypi.yml:50]: I meant switching to `build` (https://github.com/pypa/build) or poetry.

**Pull request merged**